src/sentry/integrations/gitlab/utils.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime
+from typing import Mapping, Optional
+
+from sentry.shared_integrations.response.base import BaseApiResponse
+
+API_VERSION = "/api/v4"
+
+
+class GitLabRateLimitInfo:
+    def __init__(self, info: Mapping[str, int]) -> None:
+        self.limit = info["limit"]
+        self.remaining = info["remaining"]
+        self.reset = info["reset"]
+        self.used = info["used"]
+
+    def next_window(self) -> str:
+        return datetime.utcfromtimestamp(self.reset).strftime("%H:%M:%S")
+
+    def __repr__(self) -> str:
+        return f"GitLabRateLimitInfo(limit={self.limit},rem={self.remaining},reset={self.reset}),used={self.used})"
+
+
+class GitLabApiClientPath:
```

Comment:
Why move this stuff out of the client?

Reply:
Ah yes, forgot to call this out. I wanted to put all my blame logic in `blame.py`, so to avoid a circular import I moved some things that didn't need to be inside `client.py` to this utils file.